src/main/java/fiveavian/proxvc/gui/AudioInputDeviceComponent.java

```diff
@@ -0,0 +1,80 @@
+package fiveavian.proxvc.gui;
+
+import fiveavian.proxvc.ProxVCClient;
+import fiveavian.proxvc.vc.AudioInputDevice;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.options.components.ButtonComponent;
+import net.minecraft.client.option.StringOption;
+
+import java.util.Arrays;
+
+public class AudioInputDeviceComponent extends ButtonComponent {
+    private final StringOption option;
+    private final GuiButton button;
+
+    private int currentSelection;
+
+    private final ProxVCClient client;
+    private String[] specifiers;
+
+    //little mix of GuiVCOptions and StringOptionComponent
+    public AudioInputDeviceComponent(ProxVCClient client, StringOption option) {
+        super("options." + option.name);
+        this.client = client;
+        this.specifiers = AudioInputDevice.getSpecifiers();
+        this.option = option;
+        this.button = new GuiButton(0, 0, 0, 150, 20, option.value);
+    }
+
+    protected void buttonClicked(int mouseButton, int x, int y, int width, int height, int relativeMouseX, int relativeMouseY) {
```

Comment:
when you override a method, an `@Override` annotation is expected on the line before.